manual/root_io/index.md

```diff
@@ -1,348 +1,222 @@
+Sometimes it is desirable to create a dictionary for everything defined in a given header file.To that end, the following directive is available:
 
 {% highlight C++ %}
-   MyClass obj;
-   auto file = TFile::Open("out.root", "RECREATE");
-   file->WriteObject(&obj, "myObj");
+#pragma link C++ defined_in "path/to/MyHeader.h";
 {% endhighlight %}
 
-For this to work, ROOT needs to know about the `MyClass` type: its data members, base classes, how to construct such an object when reading it back, etc.
-This is provided through a [dictionary]({{'/manual/root_io/#generating-dictionaries' | relative_url }}).
+Make sure that `path/to/MyHeader.h` corresponds to one of the header files that is passed to the `rootcling` invocation.
 
-#### Constructors
 
-ROOT's I/O function requires that each class have one of the following constructors:
+## Choosing between row-wise and columnar storage
 
-  - **Default constructor**<br>
-   A public constructor with zero parameters or with one or more parameters that all have default values: `MyClass(int = 42);`.
+ROOT data is very often stored inside {% include ref class="TTree" %} objects (which are in turn stored inside ROOT files, often manipulated via the {% include ref class="TFile" %} class), but it is also possible to store your custom types directly inside a TFile. To pick one or the other option, think of TFiles as directories and TTrees as databases or datasets: if you want to save a single object to a ROOT file, you can store it directly in the TFile (e.g. via TFile::WriteObjectAny); if you want to store several different values of a given type and later access all of those values as part of a single dataset/database, then it's probably better to use a TTree.
 
-  - **I/O constructor**<br>
-   A constructor with exactly one parameter of type "pointer to `ioctortype`".
+For more information on ROOT files, see [ROOT files]({{ '/manual/root_files' | relative_url }}).<br>
+For more information on TTree, see [Trees]({{ '/manual/trees' | relative_url }}).
 
-{% highlight C++ %}
-   struct ioctortype;
-   class MyKlass {
-   protected:
-     MyClass() = default;    // Protected: Cannot be used by ROOT I/O.
-   public:
-     MyClass(ioctortype*);
-    ...
-{% endhighlight %}
 
-#### Member initialization
+## Restrictions on types ROOT I/O can handle
 
-When you set the values of your object, ROOT also sets pointers.
-If a pointer is non-null, it must assume that the object is pointing to something and will delete object it is pointing to.
-To avoid a crash here, make sure you initialize your pointers in the constructor used by the I/O - the default constructor or the I/O constructor.
+For ROOT to be able to store a class, it must have a public constructor.
+
+ROOT currently does not support I/O of `std::shared_ptr`, `std::optional`, `std::variant` and classes with data members of these types.
```

Comment:
```
and classes with data members of these types.
```
replaced by
```
and classes with "persistent" data members of these types.
```

Reply:
done in https://github.com/root-project/web/pull/735